torchtext/data/utils.py

```diff
@@ -101,8 +101,6 @@ def get_tokenizer(tokenizer, language='en'):
         return _split_tokenizer
 
     if tokenizer == "basic_english":
-        if language != 'en':
```

Comment:
This change should be isolated to the spacy tokenizer, so we should change the meaning of the ```language``` argument to the spacy part of this function and not change the default value and its meaning for all types of tokenizers.

For BC compatibility we could add a check to the ```spacy``` part of below tokenizer that appends ```_core_web_sm``` is the language is ```en``` and throws a warning that references this PR indicating to users that now need to use this name for the languages.